### DISCUSSION TITLE
How would you write a DAX measure to display the time it takes to render visual

### DISCUSSION THREAD

**Discussion by TinyTimothy2** [TabularEditor/TabularEditor3#1253] [Data Modeling]
How would you write a DAX measure to display the time it takes to render visual
Essentially display in seconds on the power bi page what is showing in performance analyzer.

**Comment by otykier** [comment]
I don’t believe there’s any way to access performance measurements using DAX.

There are ways to obtain the performance metrics, but you have to resort to other techniques to extract them: https://learn.microsoft.com/en-us/power-bi/guidance/monitor-report-performance#monitor-premium-metrics

**Reply by TinyTimothy2** [reply]
Thanks.  My goal was to show on page what a difference implementing xyz modeling or dax change has.
Have them run side by side and show the perf differences.
(All SSAS, PB front end.  So nothing to do with staging, just dax)
I thought maybe there was a way to tag the START and END.
I just need seconds (not all the other interesting stats on SE vs FE , number of queries etc)
Seems like you could wrap start and stop around somethng.
Maybe something like this?? note this does NOT work

Visual Render Time =
VAR _StartTime = NOW()
CALCULATE (
    COUNTROWS ( 'YourTableNameHere' ),  -- Replace 'YourTableNameHere' with the name of your table
    ALLSELECTED()
) + 0   // To force evaluation of the measure
VAR _EndTime = NOW()  
VAR _RenderTime = _EndTime - _StartTime
RETURN
    _RenderTime
